interpreter/attribute_patterns_test.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package interpreter
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/google/cel-go/common/packages"
+	"github.com/google/cel-go/common/types"
+)
+
+type attr struct {
+	maybe bool
+	name  string
+	quals []interface{}
+}
+
+type patternTest struct {
+	pattern *AttributePattern
+	pkg     string
+	matches []attr
+	misses  []attr
+}
+
+var patternTests = map[string]patternTest{
+	"var": {
+		pattern: NewAttributePattern("var"),
+		matches: []attr{
+			{name: "var"},
+			{name: "var", quals: []interface{}{"field"}},
+		},
+		misses: []attr{
+			{name: "ns.var"},
+		},
+	},
+	"var_namespace": {
+		pattern: NewAttributePattern("ns.app.var"),
+		pkg:     "ns.app",
+		matches: []attr{
+			{name: "ns.app.var"},
+			{name: "ns.app.var", quals: []interface{}{int64(0)}},
+			{name: "ns", quals: []interface{}{"app", "var", "foo"}, maybe: true},
+		},
+		misses: []attr{
+			{name: "ns.var"},
+			{name: "ns", quals: []interface{}{"var"}, maybe: true},
+		},
+	},
+	"var_field": {
+		pattern: NewAttributePattern("var").Field("field"),
+		matches: []attr{
+			{name: "var"},
```

Comment:
Make sure that it's clear in the comments in the `cel` package that pattern `a.b` also matches `a` but not `a.c`.